src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java

```diff
@@ -199,7 +202,7 @@ public String getPilotCourtId(HearingCourt hearingCourt) {
 
         return pilotCourts.keySet()
             .stream()
-            .filter(pilotCourtId ->  getPilotHearingCourt(pilotCourtId).equals(hearingCourt))
+            .filter(pilotCourtId ->  getPilotHearingCourt(pilotCourtId).get().equals(hearingCourt))
```

Comment:
.get() may have trouble passing a gate. Pls verify.

Reply:
It passes :)

Reply:
How about?
```
getPilotHearingCourt(pilotCourtId).filter(hearingCourt::equals).isPresent())
```

Reply:
if `getPilotHearingCourt(pilotCourtI` going to always return an non empty object, then no need to wrap it in optional. Otherwise .get() call is dangerous.